imap/mboxlist.c

```diff
@@ -1895,22 +1895,32 @@ EXPORTED int mboxlist_createmailbox(const mbentry_t *mbentry,
     else if (config_getswitch(IMAPOPT_MAILBOX_LEGACY_DIRS))
         mbtype |= MBTYPE_LEGACY_DIRS;
 
+    newmbentry = mboxlist_entry_create();
+    newmbentry->acl = xstrdupnull(acl);
+    newmbentry->mbtype = mbtype | MBTYPE_RESERVE;
+    newmbentry->partition = xstrdupnull(newpartition);
+    newmbentry->uniqueid = xstrdup(uniqueid ? uniqueid : makeuuid());
```

Comment:
this is hoisting the "create the UUID" out of mailbox_create below.  Are there any remaining callers of mailbox_create that don't have a uniqueid?  If not, we should assert for it inside and remove the conditional from there.

Reply:
Done